### ISSUE TITLE
pg-promise breaks in ES6

### ISSUE THREAD

**Issue by kdex** [vitaly-t/pg-promise#51]
Note that currently, `index.js` makes use of an implicit global named `queryResult` and is, thus, not in strict mode. However, this completely breaks the potential to use `pg-promise` from an ES6 environment (for example within a module). Note that ES6 modules are **always** strict according to the spec.

Transpilers such as `babel` will therefore implicitly set all modules to strict mode which forbids implicit globals. This means that transpiling this:

```js
import pgp from "pg-promise";
```

will simply throw:

```js
/home/kdex/dev/chat/node_modules/pg-promise/lib/index.js:19
queryResult = {
            ^

ReferenceError: queryResult is not defined
    at Object.<anonymous> (/home/kdex/dev/chat/node_modules/pg-promise/lib/index.js:17:12)
    at Module._compile (module.js:434:26)
    at normalLoader (/usr/lib/node_modules/babel/node_modules/babel-core/lib/api/register/node.js:199:5)
    at Object.require.extensions.(anonymous function) [as .js] (/usr/lib/node_modules/babel/node_modules/babel-core/lib/api/register/node.js:216:7)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/home/kdex/dev/chat/src/server/js/db.js:5:30)
    at Module._compile (module.js:434:26)
```

If the code **must** use globals, couldn't it just reference `global.queryResult`? This should work just fine, shouldn't it? This way, we could support newer environments, too.

**Comment by vitaly-t** [comment]
Duplicate of #41

This has been already changed in 2.0 branch, which will be released just as it's ready ;)

**Comment by kdex** [comment]
Ah, that's great news. Much obliged!

**Comment by vitaly-t** [comment]
@kdex just so, 2.0.5 (pre-release) is available, and it is stable. And if you are stuck with that issue you described, you can try using that one instead. I just need to update lots of documentation before making an official release.

**Comment by kdex** [comment]
Thanks for the update, @vitaly-t. But out of curiosity, are you certain that [2.0.6 (pre-release)](https://github.com/vitaly-t/pg-promise/releases/tag/v.2.0.6) has been shipped correctly? For me, after inspecting the archive, the `package.json` still shows:

```json
    "version": "1.11.0",
```

and the implicit global `queryResults` is still being used in `lib/index.js`. Is this intended?

**Comment by vitaly-t** [comment]
2.0 is a pre-release, so you can only install it like this:
```
npm install pg-promise@2.x
```

**Comment by kdex** [comment]
Thanks a lot for the pointer, but I was actually referencing the downloadable `zip`/`tar` archives that this repository links to on the bottom [this](https://github.com/vitaly-t/pg-promise/releases/tag/v.2.0.6] page which makes this `package.json` fail:

```
    "pg-promise": "https://github.com/vitaly-t/pg-promise/archive/v.2.0.6.tar.gz",
```

Anyway, I just ran `babel` against `2.0.6`, and it worked flawlessly. Thank you!

**Comment by vitaly-t** [comment]
That's strange, because those links are automatic, provided by GitHub, and they do not fail for me here.

**Comment by kdex** [comment]
Let me rephrase: By *failing*, I was referring to them pointing to the contents of the wrong version:

```bash
$ wget https://github.com/vitaly-t/pg-promise/archive/v.2.0.6.tar.gz
$ tar -xf v.2.0.6.tar.gz 
$ cat pg-promise-v.2.0.6/package.json | grep version
  "version": "1.11.0",
$
```

**Comment by vitaly-t** [comment]
You are right, this looks like a bug on GitHub, because those releases themselves were done correctly.

At least the npm command works correctly.

**Comment by vitaly-t** [comment]
**UPDATE:** I think I know what went wrong there. I forgot to change the destination branch to point to 2.0 when doing those pre-releases. My bad. Will pay attention for the next update. Thank you for pointing this out!

:+1:

**Comment by vitaly-t** [comment]
@kdex pre-release 2.0.7 is now available, which no longer has the issues with those links. I also deleted the previous pre-releases, as they weren't pre-released correctly.

**Comment by kdex** [comment]
@vitaly-t Ah, neat! Thanks a lot for all your effort! Everything works as expected. :smiley_cat:

**Comment by vitaly-t** [comment]
@kdex You are welcome! And [version 2.0 is finally out!](https://github.com/vitaly-t/pg-promise/releases/tag/v.2.0.8) :)
